packages/nimble-components/src/table/index.ts

```diff
@@ -101,21 +100,30 @@ export class Table<
     }
 
     private trySetData(newData: TData[]): void {
-        const areIdsValid = this.tableValidator.validateDataIds(
+        const areIdsValid = this.tableValidator.validateRecordIds(
             newData,
             this.idFieldName
         );
         if (areIdsValid) {
-            this.updateTableOptions({ data: newData });
+            this.updateTableOptions({
+                data: newData,
+                getRowId: this.getRowIdFunction
```

Comment:
do we need to save the `getRowIdFunction` on the table as a property as a result of `idFieldNameChanged`? Can we just set it here after doing validation?

Reply:
No need to set it to a property. I can do everything locally within this function.